Sources/TuistKit/Commands/FocusCommand.swift

```diff
@@ -46,60 +34,52 @@ class FocusCommand: NSObject, Command {
         let modelLoader = GeneratorModelLoader(fileHandler: fileHandler,
-                                                    manifestLoader: GraphManifestLoader())
+                                                    manifestLoader: manifestLoader)
+        let generator = Generator(system: system,
+                                  printer: printer,
+                                  fileHandler: fileHandler,
+                                  modelLoader: modelLoader)
         self.init(parser: parser,
-                  graphLoader: GraphLoader(modelLoader: modelLoader),
-                  workspaceGenerator: WorkspaceGenerator(),
-                  printer: Printer(),
-                  system: System(),
-                  resourceLocator: ResourceLocator(),
+                  generator: generator,
                   fileHandler: fileHandler,
+                  manifestLoader: manifestLoader,
                   opener: Opener())
     }
 
     /// Initializes the focus command with its attributes.
     ///
     /// - Parameters:
     ///   - parser: Argument parser that parses the CLI arguments.
-    ///   - graphLoader: Graph loader instance to load the dependency graph.
-    ///   - workspaceGenerator: Workspace generator instance to generate the project workspace.
-    ///   - printer: Printer instance to output messages to the user.
-    ///   - system: System instance to run commands on the system.
-    ///   - resourceLocator: Resource locator instance used to find files in the system.
+    ///   - generator: Generator instance to generate the project workspace.
     ///   - fileHandler: File handler instance to interact with the file system.
+    ///   - manifestLoader: Manifest loader instance that can load project maifests from disk
     ///   - opener: Opener instance to run open in the system.
-    ///   - graphUp: Graph up instance to print a warning if the environment is not configured at all.
     init(parser: ArgumentParser,
-         graphLoader: GraphLoading,
-         workspaceGenerator: WorkspaceGenerating,
-         printer: Printing,
-         system: Systeming,
-         resourceLocator: ResourceLocating,
+         generator: Generating,
          fileHandler: FileHandling,
+         manifestLoader: GraphManifestLoading,
          opener: Opening) {
-        let subParser = parser.add(subparser: FocusCommand.command, overview: FocusCommand.overview)
-        self.graphLoader = graphLoader
-        self.workspaceGenerator = workspaceGenerator
-        self.printer = printer
-        self.system = system
-        self.resourceLocator = resourceLocator
+        parser.add(subparser: FocusCommand.command, overview: FocusCommand.overview)
+        self.generator = generator
         self.fileHandler = fileHandler
+        self.manifestLoader = manifestLoader
         self.opener = opener
-        configArgument = subParser.add(option: "--config",
```

Comment:
This wasn't strictly related to the change - happy to revert it and do a separate PR, thought since I was in the area to make a small improvement - as far as I can see `--config` option isn't used. (Should I add a CHANGELOG entry to state the `--config` option has been removed?)